container.go

```diff
@@ -50,6 +53,20 @@ type Container struct {
 	graph.Graph
 }
 
+// InvokeOnce only allows function invocation once to register the
+// return types. If return types are already registered, specific error
+// is returned to the caller
+func (c *Container) InvokeOnce(t interface{}) error {
+	ctype := reflect.TypeOf(t)
+	if ctype.Kind() != reflect.Func {
+		return errors.Wrapf(errParamType, _invokeErr, ctype)
+	}
+	if err := c.ValidateReturnTypes(ctype, true); err != nil {
+		return ErrInvokeOnce
+	}
+	return c.Invoke(t)
```

Comment:
gotchu. `c.InvokeOnce(newT2)` will return `dig.ErrInvokeOnce` and not Invoke `newT2`. Will add test.